ext/FHistMakieExt.jl

```diff
@@ -107,37 +131,59 @@ function Makie.plot!(input::RatioHist{<:Tuple{<:Hist1D, <:Hist1D}})
     ratiohist!(input, hratio)
 end
 
+Makie.used_attributes(::Type{<:Makie.Plot}, h::Hist1D) = (:clamp_bincounts, )
+function Makie.convert_arguments(P::Type{<:Scatter}, h::Hist1D; clamp_bincounts=false)
+    ys = copy(bincounts(h))
+    if clamp_bincounts
+        _clamp_counts!(ys)
+    end
+    convert_arguments(P, bincenters(h), ys)
+end
+function Makie.convert_arguments(P::Type{<:BarPlot}, h::Hist1D; clamp_bincounts=false)
+    ys = copy(bincounts(h))
+    if clamp_bincounts
+        _clamp_counts!(ys)
+    end
+    convert_arguments(P, bincenters(h), ys)
+end
+
 Makie.MakieCore.plottype(::Hist1D) = Hist
-function Makie.convert_arguments(P::Type{<:Stairs}, h::Hist1D)
+function Makie.convert_arguments(P::Type{<:Stairs}, h::Hist1D; clamp_bincounts=false)
     edges = binedges(h)
     phantomedge = edges[end] # to bring step back to baseline
     bot = eps()
-    bc = bincounts(h)
+    bc = copy(bincounts(h))
+    if clamp_bincounts
+        _clamp_counts!(bc)
+    end
     z = zero(eltype(bc))
     nonzero_bincounts = replace(bc, z => bot)
     convert_arguments(P, vcat(edges, phantomedge), vcat(bot, nonzero_bincounts, bot))
 end
-Makie.convert_arguments(P::Type{<:Scatter}, h::Hist1D) = convert_arguments(P, bincenters(h), bincounts(h))
-Makie.convert_arguments(P::Type{<:BarPlot}, h::Hist1D) = convert_arguments(P, bincenters(h), bincounts(h))
 
-function Makie.convert_arguments(P::Type{<:Makie.Errorbars}, h::FHist.Hist1D)
-    se = FHist.binerrors(FHist.sqrt, h)
-    bincounts = FHist.bincounts(h)
-    if all(==(1.0), FHist.sumw2(h))
-        # no weights used in filling
-        convert_arguments(P, FHist.bincenters(h), bincounts, se)
+Makie.used_attributes(::Type{<:Errorbars}, h::Hist1D) = (:clamp_bincounts, :clamp_errors, :error_function)
+function Makie.convert_arguments(P::Type{<:Makie.Errorbars}, h::FHist.Hist1D; clamp_bincounts=false, clamp_errors=true, error_function=nothing)
+    xs = FHist.bincenters(h)
+    ys = copy(FHist.bincounts(h))
+    errs = if isnothing(error_function)
+        FHist.binerrors(FHist.sqrt, h)
     else
-        # weights used, switch to pearson error if error bar dips below zero
-        pe = FHist.binerrors(FHist.pearson_err, h)
-        err_high, pe_low = first.(pe), last.(pe)
-        err_low = se
-        for i in eachindex(bincounts, se, pe_low)
-            if bincounts[i] - err_low[i] <= 0
-                err_low[i] = pe_low[i]
+        FHist.binerrors(error_function, h)
+    end
+    hi_errs, lo_errs = first.(errs), last.(errs)
+
+    if clamp_bincounts && clamp_errors
+        _clamp_counts_errors!(ys, lo_errs, hi_errs)
+    elseif clamp_bincounts && !clamp_errors
+        _clamp_counts(ys)
```

Comment:
I'm not sure if this should be allowed. Maybe it should throw a warning about possible inconsistencies on the errorbars for negative values? Or maybe it's an overkill.

Reply:
so I think we agreed we should NOT clam

Reply:
ah, I think in the examples I said "don't do this" but in the code I imssed it, will fix